src/python_testing/matter_testing_infrastructure/nfc.py

```diff
@@ -0,0 +1,318 @@
+            self.tag_monitor.deleteObserver(self.observer)
+            logger.info("Stopped monitoring NFC tags")
+
+    async def activate(self):
+        self._stop_event.clear()
+        self._thread = threading.Thread(target=self._monitoring_thread, daemon=True)
+        self._thread.start()
+        # Return control to the caller, but keep this coroutine alive until deactivation
+        while not self._stop_event.is_set():
+            await asyncio.sleep(0.1)
+        # After deactivation, join the thread to ensure cleanup
+        if self._thread is not None:
+            self._thread.join()
+            self._thread = None
+        # Return the last NDEF bytes read
+        if self.observer:
+            return self.observer.last_ndef
+        return None
+
+    def deactivate(self):
+        self._stop_event.set()
+
+
+class NFCReaderHelper:
+    """
+    Helper class for NFC reader operations and tag parsing.
+    """
+
+    def __init__(self, reader=None):
+        """
+        If reader is None, automatically connect to the first available reader.
+        """
+        if reader is None:
+            reader = self.get_connected_reader()
+            if reader is None:
+                raise RuntimeError("No smartcard reader found.")
+        self.reader = reader
+        self._monitor_manager = None
+
+    @staticmethod
+    def get_connected_reader():
+        """
+        Function to find and return the first available NFC reader.
+        """
+        reader_list = smartcard.System.readers()
+        if not reader_list:
+            logger.info("No smartcard reader found.")
+            return None
+        logger.debug("Available readers:")
+        for idx, reader in enumerate(reader_list):
+            logger.debug(f"{idx}: {reader}")
+        reader = reader_list[0]  # Use the first reader found
+        logger.debug(f"Using NFC reader: {reader}")
+        return reader
+
+    def activate_tag_monitoring(self):
+        """
+        Start monitoring in the background and return control immediately.
+        """
+        self._monitor_manager = TagMonitorManager(self)
+        return self._monitor_manager.activate()
+
+    def deactivate_tag_monitoring(self):
+        """
+        Stop tag monitoring.
+        """
+        if self._monitor_manager is not None:
+            self._monitor_manager.deactivate()
+            self._monitor_manager = None
+
+    def ndef_content_to_string(self, ndef_bytes):
+        """
+        Returns a string representation of all NDEF records and their fields.
+        """
+        if not ndef_bytes:
+            return "No NDEF data to decode."
+
+        lines = []
+        try:
+            for idx, record in enumerate(ndef.message_decoder(ndef_bytes), 1):
+                lines.append(f"NDEF Record {idx}: {record}")
+                # Following has limited added value but is kept commented for reference
+                # for attr in ['tnf', 'type', 'id', 'uri', 'text', 'data']:
```

Comment:
commented out code is generally not amazing. How about `logging.debug` these things instead?